### PR TITLE
Issue #459: Deprecate StringHistogram for backward compatibility 

### LINKED ISSUE #459: Rollback and deprecate StringHistogramTransformation
We noticed some users have tables with `StringHistogramTransformation`, which was currently transformed into `CDFQuantilesStringTransformation`. 

Although we can transform the Revision metadata in a new transaction, we should roll back the original class's code and maintain it to enable backward compatibility. 

This is a priority before releasing any new version.

### PR DESCRIPTION
**Pull request by osopardo1**
## Description

Fixes #459 

## Type of change

Bug Fix to ensure backward compatibility within new versions. 

## Checklist:

Here is the list of things you should do before submitting this pull request:

- [x] New feature / bug fix has been committed following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add logging to the code following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add comments to the code (make it easier for the community!).
- [x] Change the documentation.
- [x] Add tests.
- [x] Your branch is updated to the main branch (dependent changes have been merged).

## How Has This Been Tested? (Optional)

This will be tested with an old table written with histogram transformation and version 0.7.0 of Qbeast Spark.

### CODE REVIEW

**Comment by osopardo1** [comment]
After discussion, we agreed on:
- Allow writing with StringHistogramTransformer and StringHistogramTransformation.
- Use superseded and merge methods to change between StringHIstogram and CDF for the moment. 
- Add a method to change from histogram to cdf metadata.

File: core/src/main/scala/io/qbeast/core/transform/StringHistogramTransformation.scala [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+   * Merges two transformations. The domain of the resulting transformation is the union of this
+   *
+   * @param other
+   *   Transformation
+   * @return
+   *   a new Transformation that contains both this and other.
+   */
+  override def merge(other: Transformation): Transformation = other match {
+    case _: StringHistogramTransformation => other
+    case _ => this
```
[the remark this answers is not in the document]
Comment (osopardo1) [reply core/src/main/scala/io/qbeast/core/transform/StringHistogramTransformation.scala:100]: done

File: src/main/scala/io/qbeast/utils/QbeastUtils.scala [hunk 2]
```diff
@@ -132,4 +150,51 @@ object QbeastUtils extends Logging {
...
+        case (_ @NullToZeroTransformation, linearTransformer: LinearTransformer) =>
+          IdentityTransformation(null, linearTransformer.dataType.asInstanceOf[OrderedDataType])
+        case (otherTransformation, _) => otherTransformation
+      }
+    revision
+      .copy(columnTransformers = updatedTransformers, transformations = updatedTransformations)
+  }
+
+  @Experimental
+  def updateTransformationTypes(qbeastTable: QbeastTable): Unit = {
```
Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/utils/QbeastUtils.scala:179]: We should ensure we don't update the metadata if there are no revision changes.

**Review by Jiaweihu08 — APPROVED** [review]

File: src/main/scala/io/qbeast/utils/QbeastUtils.scala (hunk 2 above)
Comment (osopardo1) [reply src/main/scala/io/qbeast/utils/QbeastUtils.scala:190]: done in [30b3e1e](https://github.com/Qbeast-io/qbeast-spark/pull/485/commits/30b3e1e315c051bd26cfb51ec7f9b05ff9e73ba7)

**Pull request merged**